hack/validate.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -eo pipefail
+
+eval $(go env)
+PATH="$GOPATH/bin:$PATH"
+
+die() { echo "Error: ${1:-No message provided}" > /dev/stderr; exit 1; }
+
+# Always run from the repository root
+cd $(dirname "${BASH_SOURCE[0]}")/../
+
+if [[ -z $(type -P gofmt) ]]; then
+    die "Unable to find 'gofmt' binary in \$PATH: $PATH"
+fi
+
+echo "Executing go vet"
+GO111MODULE="on" go vet ./...
+
+echo "Executing gofmt"
+# grep exits non-zero when no matches found
+OUTPUT=$(gofmt -s -l . |& egrep -v '^vendor' || true)
```

Comment:
This seems to be parsed as (`gofmt -s -l . |& egrep -v '^vendor'`) `||` `true`, and e.g. invalid Gofmt flags don’t cause a failure.

Maybe
```suggestion
OUTPUT=$(gofmt -s -l . |& { egrep -v '^vendor' || true; })
```

Reply:
Hmmm, I didn't think so, but am not entirely sure now that you mention it :confused: Rather than complicate things more, it may be cleaner to just not use grep here...

Reply:
...there.  Replaced `grep` with `sed`.

Reply:
I wasn’t sure at all either, so I tested that:
```bash
$ set -o pipefail
$ gofmt -eee -s -l . 2>&1 | { cat; false; } || true; echo $?
flag provided but not defined: -eee
…
0
```
(with the `cat` to avoid a broken pipe failure).

The `sed` solution is inventive. I have learned a new trick.